src/Fixer/Phpdoc/AlignMultilineCommentFixer.php

```diff
@@ -0,0 +1,156 @@
+{
+    private $tokenKinds;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        $this->tokenKinds = [T_DOC_COMMENT];
+        if ('phpdocs_only' !== $this->configuration['comment_type']) {
+            $this->tokenKinds[] = T_COMMENT;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Each line of multi-line DocComments must have an asterisk [PSR-5] and must be aligned with the first one.',
+            [
+                new CodeSample(
+'<?php
+    /**
+            * This is a DOC Comment
+with a line not prefixed with asterisk
+
+   */'
+                ),
+                new CodeSample(
+'<?php
+    /*
+            * This is a doc-like multiline comment
+*/',
+                    ['comment_type' => 'phpdocs_like']
+                ),
+                new CodeSample(
+'<?php
+    /*
+            * This is a doc-like multiline comment
+with a line not prefixed with asterisk
+
+   */',
+                    ['comment_type' => 'all_multiline']
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound($this->tokenKinds);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind($this->tokenKinds)) {
+                continue;
+            }
+
+            $whitespace = '';
+            $previousIndex = $index - 1;
+            if ($tokens[$previousIndex]->isWhitespace()) {
+                $whitespace = $tokens[$previousIndex]->getContent();
+                --$previousIndex;
+            }
+            if ($tokens[$previousIndex]->isGivenKind(T_OPEN_TAG)) {
+                $whitespace = preg_replace('/\S/', '', $tokens[$previousIndex]->getContent()).$whitespace;
+            }
+
+            if (1 !== preg_match('/\R([ \t]*)$/', $whitespace, $matches)) {
+                continue;
+            }
+
+            if ($token->isGivenKind(T_COMMENT) && 'all_multiline' !== $this->configuration['comment_type'] && 1 === preg_match('/\R(?:\R|\s*[^\s\*])/', $token->getContent())) {
+                continue;
+            }
+
+            $indentation = $matches[1];
+            $lines = preg_split('/\R/', $token->getContent());
+            foreach ($lines as $lineNumber => $line) {
+                if (0 === $lineNumber) {
+                    continue;
+                }
+
+                $line = ltrim($line);
+                if ($token->isGivenKind(T_COMMENT) && (!isset($line[0]) || '*' !== $line[0])) {
```

Comment:
Yep, but the first check is done only once, not on each line. Maybe you could reuse the regular expression in some way that would not require to iterate each line. But I'm fine with current code TBH.